docs/manual/src/proc_macro/index.md

```diff
@@ -186,6 +186,19 @@ impl Foo {
 }
 ```
 
+Exported functions can input object arguments as either an `Arc<>` or reference.
+```rust
+// Input foo as an Arc and bar as a reference
+fn call_both(foo: Arc<Foo>, bar: &Foo) {
+  foo.method_a();
+  bar.method_rba();
+```
+
+There are a couple limitations when using references for arguments:
+  - They can only be used with interface types: objects, trait interfaces, or callback interfaces.
```

Comment:
Note: I've only actually tested this with objects, but I'm pretty sure the other two possibilities should work as well.